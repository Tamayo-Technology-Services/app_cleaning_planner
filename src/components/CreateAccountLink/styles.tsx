import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  text: {
    fontSize: 12,
    color: 'black',
    fontWeight: '300',
  },
  createAccountText: {
    top: 2,
    fontSize: 12,
    color: '#0070F4',
    fontWeight: '300',
  },
})