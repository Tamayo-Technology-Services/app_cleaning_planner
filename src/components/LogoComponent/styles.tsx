
import { StyleSheet } from 'react-native';

export const loginStyles = StyleSheet.create({
logoContainer: {
    alignItems: 'center',
    marginBottom: 30,
  },
  logoText: {
    color: 'black',
    fontSize: 48,
    fontWeight: 'bold',
  },
  logo: {
    resizeMode: 'contain',
  },
})