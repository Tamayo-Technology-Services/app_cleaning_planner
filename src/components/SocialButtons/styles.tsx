import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  socialButtonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
    marginLeft: 15,
    marginRight: 15,
  },
  socialButton: {
    alignItems: 'center',
    marginLeft: 25,
    marginRight: 10,
    backgroundColor: '#fff',
    paddingVertical: 10,
  
    borderColor: '#F1F1F1',
    borderWidth: 1,
    borderRadius: 8,
    flex: 1,
  },
  socialButtonText: {
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
  },
  socialButtonImage: {
    width: 13,
    height: 13,
    resizeMode: 'contain',
  }
})