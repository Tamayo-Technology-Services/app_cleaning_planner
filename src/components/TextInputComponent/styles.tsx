
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
input: {
    color: 'black',
    width: '100%',
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 5,
    marginBottom: 10,
    paddingHorizontal: 10,
  },
})