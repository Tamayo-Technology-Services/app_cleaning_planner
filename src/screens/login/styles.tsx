import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
    container: {
    flex: 1,
    justifyContent: 'center',
    // alignItems: 'center',
    padding: 10,
  },
  input: {
    width: '80%',
    height: 40,
    marginVertical: 10,
    borderWidth: 1,
    padding: 10,
  },
  forgotPasswordText: {
    textAlign: 'right',
    right: 20,
    marginBottom: 20,
    fontSize: 12,
    fontWeight: '300',
  },
  containerButton: {
    justifyContent: 'center', 
    alignItems: 'center'
  },
  loginButton: {
    width: '35%',
    backgroundColor: '#0070F4',
    paddingVertical: 10,
    borderRadius: 7,
  },
  loginButtonText: {
    color: 'white',
    fontSize: 12,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  orText: {
    color: 'black',
    left: 30,
    marginTop: 20,
    marginBottom: 10,
    fontSize: 12,
    fontWeight: '300'
  },
  loginText:{
    fontWeight: 'bold',
    textAlign: 'left',
    paddingLeft: 30,
    paddingBottom: 10,

  }
})
