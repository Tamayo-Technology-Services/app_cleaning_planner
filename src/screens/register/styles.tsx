import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
    container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
  },
  input: {
    width: '80%',
    height: 40,
    marginVertical: 10,
    borderWidth: 1,
    padding: 10,
  },
  forgotPasswordText: {
    textAlign: 'right',
    marginBottom: 20,
    color: 'blue',
  },
  loginButton: {
    backgroundColor: 'blue',
    paddingVertical: 15,
    paddingHorizontal: 50,
    borderRadius: 25,
  },
  loginButtonText: {
    color: 'white',
    fontSize: 18,
    textAlign: 'center',
  },
  orText: {
    color: 'black',
    marginTop: 20,
    marginBottom: 10,
    fontSize: 18,
  },
})