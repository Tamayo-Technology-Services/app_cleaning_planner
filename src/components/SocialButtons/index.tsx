import React from 'react';
import { View, TouchableOpacity, Text, StyleProp, ViewStyle } from 'react-native';
import { styles } from './styles';

interface SocialButtonsProps {
  onPressFacebook: () => void;
  onPressGoogle: () => void;
  onPressApple: () => void;
  containerStyle?: StyleProp<ViewStyle>;
}

const SocialButtons: React.FC<SocialButtonsProps> = ({ onPressFacebook, onPressGoogle, onPressApple, containerStyle }) => {
  return (
    <View style={[styles.socialButtonContainer, containerStyle]}>
      <TouchableOpacity style={styles.socialButton} onPress={onPressFacebook}>
        <Text style={styles.socialButtonText}>Facebook</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.socialButton} onPress={onPressGoogle}>
        <Text style={styles.socialButtonText}>Google</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.socialButton} onPress={onPressApple}>
        <Text style={styles.socialButtonText}>Apple</Text>
      </TouchableOpacity>
    </View>
  );
};

export default SocialButtons;