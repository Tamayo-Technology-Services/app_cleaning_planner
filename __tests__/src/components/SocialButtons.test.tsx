import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import SocialButtons from '../../../src/components/SocialButtons';
import { styles } from '../../../src/components/SocialButtons/styles';

describe('SocialButtons', () => {
  test('renders correctly', () => {
    const { getByText } = render(
      <SocialButtons onPressFacebook={() => {}} onPressGoogle={() => {}} onPressApple={() => {}} />
    );
    expect(getByText('Facebook')).toBeTruthy();
    expect(getByText('Google')).toBeTruthy();
    expect(getByText('Apple')).toBeTruthy();
  });

  test('matches snapshot', () => {
    const { toJSON } = render(
      <SocialButtons onPressFacebook={() => {}} onPressGoogle={() => {}} onPressApple={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('calls onPress functions correctly', () => {
    const onPressFacebookMock = jest.fn();
    const onPressGoogleMock = jest.fn();
    const onPressAppleMock = jest.fn();

    const { getByText } = render(
      <SocialButtons
        onPressFacebook={onPressFacebookMock}
        onPressGoogle={onPressGoogleMock}
        onPressApple={onPressAppleMock}
      />
    );

    fireEvent.press(getByText('Facebook'));
    fireEvent.press(getByText('Google'));
    fireEvent.press(getByText('Apple'));

    expect(onPressFacebookMock).toHaveBeenCalled();
    expect(onPressGoogleMock).toHaveBeenCalled();
    expect(onPressAppleMock).toHaveBeenCalled();
  });
});