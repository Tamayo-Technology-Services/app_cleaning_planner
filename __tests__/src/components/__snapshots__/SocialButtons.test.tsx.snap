// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtons matches snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#3b5998",
        "borderRadius": 25,
        "flex": 1,
        "marginRight": 10,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Facebook
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#3b5998",
        "borderRadius": 25,
        "flex": 1,
        "marginRight": 10,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Google
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#3b5998",
        "borderRadius": 25,
        "flex": 1,
        "marginRight": 10,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Apple
    </Text>
  </View>
</View>
`;
